### PR TITLE
pkg/kubelet: Optimize switch code

### PR DESCRIPTION
**Pull request by zhouhao3**
Signed-off-by: Zhou Hao <zhouhao@cn.fujitsu.com>

**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Optimize switch code

### CODE REVIEW

File: pkg/kubelet/kubelet_node_status.go [hunk 1]
```diff
@@ -599,9 +599,7 @@ func validateNodeIP(nodeIP net.IP) error {
 	for _, addr := range addrs {
 		var ip net.IP
 		switch v := addr.(type) {
-		case *net.IPNet:
-			ip = v.IP
-		case *net.IPAddr:
+		case *net.IPNet, *net.IPAddr:
```
Comment (mattjmcnaughton) [inline pkg/kubelet/kubelet_node_status.go:602]: Could you share more about the advantages of a `case` statement vs a single `if` statement, given that we only have a single block in our case statement?

**Review by mattjmcnaughton — COMMENTED** [review]
Welcome to k8s and thanks for your pr!

File: pkg/kubelet/kubelet_node_status.go (hunk 1 above)
Comment (q384566678) [reply pkg/kubelet/kubelet_node_status.go:602]: My change here is to merge two cases with the same treatment into one. As for the comparison with if, I think it is more concise than the code implemented by if.

Comment (mattjmcnaughton) [reply pkg/kubelet/kubelet_node_status.go:602]: Ok - curious if other have strong opinions. At the end of the day, I think both are better than a switch statement with two duplicate cases.

Do you have thoughts on the error message we are getting when running `govet`? 

```
pkg/kubelet/kubelet_node_status.go:603:10: v.IP undefined (type "net".Addr has no field or method IP)
```

From https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/90100/pull-kubernetes-verify/1249579974818533377.

Comment (q384566678) [reply pkg/kubelet/kubelet_node_status.go:602]: I did not notice this problem before, it seems that this problem will not occur before the change. This is very strange, and I don’t quite understand the cause of this problem.

Comment (q384566678) [reply pkg/kubelet/kubelet_node_status.go:602]: After understanding the go `type switch`, I think this modification is wrong, the original code does not need to be modified.

**Pull request closed without merging**